core/templates/dev/head/components/HintAndSolutionButtonsDirective.js

```diff
@@ -0,0 +1,95 @@
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Directive for hint and solution buttons.
+ */
+
+oppia.directive('hintAndSolutionButtons', [
+  'UrlInterpolationService', function(UrlInterpolationService) {
+    return {
+      restrict: 'E',
+      scope: {},
+      templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+        '/components/hint_and_solution_buttons_directive.html'),
+      controller: [
+        '$scope', '$rootScope', 'HintManagerService',
+        'ExplorationPlayerService', 'SolutionManagerService',
+        'PlayerTranscriptService', 'HintAndSolutionModalService',
+        'DeviceInfoService', 'WindowDimensionsService',
+        function(
+          $scope, $rootScope, HintManagerService,
+          ExplorationPlayerService, SolutionManagerService,
+          PlayerTranscriptService, HintAndSolutionModalService,
+          DeviceInfoService, WindowDimensionsService) {
+          $scope.defaultHintUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/default_hint.svg');
+
+          $scope.activeHintUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/active_hint.svg');
+
+          $scope.defaultSolutionUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/default_solution.svg');
+
+          $scope.activeSolutionUrl = UrlInterpolationService.getStaticImageUrl(
+            '/icons/active_hint.svg');
+
+          $scope.hintPositions = [];
+          $scope.solutionExists = null;
+          $scope.isWindowNarrow = WindowDimensionsService.isWindowNarrow();
+          $scope.activeModalIndex = null;
+
+          $rootScope.$on('hintsAndSolutionReset', function(evt, data) {
+            for(var index = 0; index < data.numOfHints; index++) {
+              $scope.hintPositions.push(index);
+            }
+            $scope.solutionExists = data.solutionExists;
+          });
+
+          $scope.isHintButtonVisibleAtIndex = function(index) {
+            if (index === 0) {
+              return HintManagerService.isCurrentHintAvailable() ||
+                index < HintManagerService.getCurrentHintIndex();
+            } else if (index === HintManagerService.getCurrentHintIndex()) {
+              return HintManagerService.isCurrentHintAvailable();
+            } else {
+              return index < HintManagerService.getCurrentHintIndex();
+            }
+          };
+
+          $scope.isSolutionButtonVisible = function() {
+            return SolutionManagerService.isCurrentSolutionAvailable() &&
+                HintManagerService.areAllHintsExhausted();
+          };
+
+          $scope.displayHintModalForIndex = function(index) {
```

Comment:
Suggest renaming: $scope.displayHintModal = function(hintIndex)

Reply:
I don't understand -- I don't see that function in this file? I'm just talking about renaming this function name, not changing any of the implementation details.

But if you still want to keep the names distinct, I'm OK with that.